corehq/apps/app_manager/util.py

```diff
@@ -507,84 +508,102 @@ def get_sort_and_sort_only_columns(detail, sort_elements):
     return sort_only_elements, sort_columns
 
 
-def get_form_data(domain, app, include_shadow_forms=True):
-    from corehq.apps.reports.formdetails.readable import FormQuestionResponse
-    from corehq.apps.app_manager.models import ShadowForm
-
-    case_meta = app.get_case_metadata()
-    modules = []
-    errors = []
-    for module in app.get_modules():
-        forms = []
-        module_meta = {
+class FormDataGenerator(object):
+    def __init__(self, domain, app, include_shadow_forms=True):
+        self.domain = domain
+        self.app = app
+        self.include_shadow_forms = include_shadow_forms
+
+        self.errors = []
+
+        self._seen_save_to_case = defaultdict(list)
+        self._case_meta = self.app.get_case_metadata()
+
+    def get_app_data(self):
+        return [self._compile_module(module) for module in self.app.get_modules()], self.errors
```

Comment:
One nitpick with this is `compile_x`, `get_x_data`, and `x.to_json` are all used in this class for essentially the same thing - would be nice to align that terminology.